src/main/java/de/tum/in/www1/artemis/domain/Result.java

```diff
@@ -105,6 +106,13 @@
     @Column(name = "example_result")
     private Boolean exampleResult;
 
+    // This attribute is required to forward the coverage file reports after creating the build result. This is required in order to
+    // delay referencing the corresponding test cases from the entries because the test cases are not saved in the database
+    // at this point of time but the required test case name would be lost, otherwise.
+    @Transient
+    @JsonIgnore
+    private Map<String, Set<CoverageFileReport>> fileReportsByTestCaseName;
```

Comment:
Is this really necessary here? Results should stay as independent as possible from specific exercise types.
Could we instead attach this to a ProgrammingSubmission?

And is it necessary to store a map? can we not store this in a more object-oriented way?

Reply:
> Results should stay as independent as possible from specific exercise types.

You could argue that it is independent from the specific exercise type. Coverage analysis is currently only available for Java/Gradle-exercises but is designed to also work for other exercise types. The `Result` also stores a list of objects of type `Feedback` which represents the result of the static code analysis. I think both scenarios are comparable in context.

> Could we instead attach this to a ProgrammingSubmission?

The `ProgrammingSubmission` will be created upon the `Result`. I think using the submission would cause a mixing of the data layer (DTOs) and domain layer (entities). The only point where the BambooResultNotificationDTO (data layer) is parsed into entities is when creating the `Result`. And if we now attach it to the ProgrammingSubmission we would have to again access the DTOs for this purpose.
And in addition, the purpose of using the Result object is to represent a build result for a programming exercise. The submission references this result. I think storing the `coverage analysis report result` into a `ProgrammingSubmission` would break the purpose of the distinction between the objects.

> can we not store this in a more object-oriented way?

I agree that this is a cumbersome way. This is due to the fact that the test cases, which are referenced by the `TestwiseCoverageReportEntry` inside of the `CoverageFileReport`, are created after the Result has been created. If the test name would not be saved, we would lose this crucially required information.
One possibility would be to create dummy test cases (with only hold the parsed test name) for the report and resolve it after the creation of the test cases entities to these entities. This would change the above type to a `CoveragReport`. But I am personally not sure if this is actually a better solution. What do you think?